### PR TITLE
[Win32SS][GDI] Fix pool memory disclosure in NtGdiGetGlyphOutline

### PR DESCRIPTION
**Pull request by khanhnt2**
## Purpose

Fix pool memory disclosure in NtGdiGetGlyphOutline.

## Analysis

The value of `pvBuf` will be initialized in function `ftGdiGetGlyphOutline`. But in some cases like https://github.com/reactos/reactos/blob/52fb8c1a8d1d1ef2ddb46c017b461011b26188ee/win32ss/gdi/ntgdi/freetype.c#L4016-L4017 `ftGdiGetGlyphOutline` will return without initializing `pvBuf` so it will lead to memory disclosure when execute this https://github.com/reactos/reactos/blob/50510538db49b58b3364a0938a82879bc08890a3/win32ss/gdi/ntgdi/font.c#L767-L768

I guess this is a bug pattern when initializer function does a comparison then return immediately without initializing.
## TODO

~~- [ ] Remove redundant RtlZeroMemory functions in `ftGdiGetGlyphOutline`~~

### CODE REVIEW

**Comment by khanhnt2** [comment]
I doubt that my fix is correct. Should I use RtlZeroMeory in the function `ftGdiGetGlyphOutline` to fix? For example, the fix will look like

```
 if (needed > cjBuf) {
     RtlZeroMemory(pvBuf, cjBuf); 
     return GDI_ERROR; 
 }
```

But there are a lot of comparison like that in the function `ftGdiGetGlyphOutline`.

File: win32ss/gdi/ntgdi/font.c [hunk 1]
```diff
@@ -749,6 +749,7 @@ NtGdiGetGlyphOutline(
         EngSetLastError(ERROR_NOT_ENOUGH_MEMORY);
         goto Exit;
      }
+     RtlZeroMemory(pvBuf, cjBuf);
```
Comment (Extravert-ir) [inline win32ss/gdi/ntgdi/font.c:752]: Let's use `ExAllocatePoolZero` since we have it in our sdk now

Comment (khanhnt2) [reply win32ss/gdi/ntgdi/font.c]: Thanks for your advice, I have changed that function.

**Comment by khanhnt2** [comment]
I realize that ftGdiGetGlyphOutline is not only used by NtGdiGetGlyphOutline, so I guess I shouldn't remove redundant RtlZeroMemory in ftGdiGetGlyphOutline.

**Review by learn-more — APPROVED** [review]

**Pull request merged**